apps/gql/loaders/user.ts

```diff
@@ -0,0 +1,27 @@
+import { createPrismaLoader } from "@kampus/gql-utils";
+
+import { type Clients } from "~/clients";
+import { LoaderKey } from "./utils/loader-key";
+
+export class UserLoaderKey extends LoaderKey<"id" | "username", string> {}
+
+export function createUserLoaders(clients: Clients) {
+  const byID = createPrismaLoader(clients.prisma.user, "id", (users) => {
+    users.forEach((user) => {
+      if (user.username) {
+        byUsername.prime(user.username, user);
+      }
+    });
+  });
+
+  const byUsername = createPrismaLoader(clients.prisma.user, "username", (users) => {
+    users.forEach((user) => {
+      byUsername.prime(user.id, user);
+    });
+  });
```

Comment:
I still don't get it how / why this "variable hoisting works before defining it 🤯.  (just a comment nothing expected as comment or resolving)

Reply:
magic of javascript. but this is actually exactly the same as defining your helper functions below your components